### PR TITLE
CB-10679: Documenting how the CLI chooses plugin versions

### PR DESCRIPTION
**Pull request by riknoll**
Now in the README! Note that this relies on the documentation in https://github.com/apache/cordova-docs/pull/559 being merged.

@nikhilkh @dblotsky please take a look.

### CODE REVIEW

**Comment by dblotsky** [comment]
LGTM.

File: doc/readme.md [hunk 1]
```diff
@@ -337,6 +337,18 @@ There are a number of ways to specify a plugin:
...
+### Algorithm for choosing which plugin version to fetch from npm
+
+When adding a plugin to a project from npm, the CLI will choose which version
+of the plugin to fetch based on the following criteria (listed in order of
+precedence):
+
+1. The plugin version specified in the command (e.g. `cordova plugin add plugin@version`)
+2. The plugin version saved in config.xml (i.e. if the plugin was previously added with `--save`)
+3. The latest plugin version that the current project can support (only applies to plugins that list their [Cordova dependencies] in their `package.json`)
+4. The plugin version pinned by your installed CLI version (only applies to [Apache-developed plugins])
```
Comment (nikhilkh) [inline doc/readme.md]: This will likely need an update in a couple of days - when you remove plugin pinning.

File: doc/readme.md [hunk 2]
```diff
@@ -337,6 +337,18 @@ There are a number of ways to specify a plugin:
 | commit-ish  | Commit/tag/branch reference. If none is specified, 'master' is used
 | subdir      | Sub-directory to find plugin.xml for the specified plugin.
 
+### Algorithm for choosing which plugin version to fetch from npm
+
+When adding a plugin to a project from npm, the CLI will choose which version
+of the plugin to fetch based on the following criteria (listed in order of
+precedence):
+
+1. The plugin version specified in the command (e.g. `cordova plugin add plugin@version`)
```
Comment (nikhilkh) [inline doc/readme.md]: Consider changing `version` to `plugin-spec`

File: doc/readme.md [hunk 3]
```diff
@@ -337,6 +337,18 @@ There are a number of ways to specify a plugin:
...
+2. The plugin version saved in config.xml (i.e. if the plugin was previously added with `--save`)
+3. The latest plugin version that the current project can support (only applies to plugins that list their [Cordova dependencies] in their `package.json`)
```
Comment (nikhilkh) [inline doc/readme.md]: "As for version 6.1, ..."

**Comment by riknoll** [comment]
@nikhilkh responded to your review. I removed the bit about plugin pinning, but do we need to explain that Cordova version 6.0.0 had it?

**Comment by nikhilkh** [comment]
LGTM. Thanks!

**Pull request closed without merging**